src/debug/shared/dbgtransportsession.cpp

```diff
@@ -1107,6 +1107,9 @@ HRESULT DbgTransportSession::CheckBufferAccess(__in_ecount(cbBuffer) PBYTE pbBuf
         return HRESULT_FROM_WIN32(ERROR_ARITHMETIC_OVERFLOW);
     }
 
+    // TODO: VirtualQuery from PAL doesn't seem to provide correct result for DAC globals on Linux.
```

Comment:
This is by design. VirtualQuery only works well for memory allocated via VirtualAlloc.